Sources/Workspace/Workspace.swift

```diff
@@ -1696,12 +1697,18 @@ extension Workspace {
                 targetName: artifact.targetName
             ]
 
-            if let existingArtifact = existingArtifact, case .remote = existingArtifact.source {
-                // If we go from a remote to a local artifact, we can remove the old remote artifact.
-                artifactsToRemove.append(existingArtifact)
+            if artifact.path.extension == "zip" {
+                // TODO: We could check here a file signature of the `existingArtifact`, and
+                // if it matches the manifest `artifact` signature we could just `continue`.
+                artifactsToExtract.append(artifact)
```

Comment:
If we want to address the `todo:` that most probably would require persisting the `existingArtifact` file signature (either a checksum or modification date) in the workspace state.

Reply:
adding that to workspace artifacts state makes sense to me

Reply:
[Done](https://github.com/apple/swift-package-manager/pull/3746/commits/4f581685bb840a8f7a2973f364ce8350db779687), I introduced the checksum as the local source case associated value. On the one hand, I like the symmetry with the remote source case, but on the other, the optional checksum requires some additional handling. Would it work for you like that, or would you prefer to iterate on that by storing the checksum in a separate `localArchived` case, renaming the associated value name to `archiveChecksum`, or maybe something else?

Reply:
I think this is good